### ISSUE TITLE
How to train my model in fp16?

### ISSUE THREAD

**Issue by SinMxchine** [cocktailpeanut/fluxgym#222]
I noticed in the training script it say's "--fp8_base^" after I trained a quick model and tried it in comfy I got the model weight errors when running flux in fp16. Does anyone know what setting I use to change the base? I don't want to train a fp8 model.

I enabled the last 2 options at the bottom but with these on, the training won't start and IDK if these change it to work with fp16.

Thank you for any help.

`ccelerate launch ^
  --mixed_precision bf16 ^
  --num_cpu_threads_per_process 1 ^
  sd-scripts/flux_train_network.py ^
  --pretrained_model_name_or_path "C:\pinokio\api\fluxgym.git\models\unet\flux1-dev.sft" ^
  --clip_l "C:\pinokio\api\fluxgym.git\models\clip\clip_l.safetensors" ^
  --t5xxl "C:\pinokio\api\fluxgym.git\models\clip\t5xxl_fp16.safetensors" ^
  --ae "C:\pinokio\api\fluxgym.git\models\vae\ae.sft" ^
  --cache_latents_to_disk ^
  --save_model_as safetensors ^
  --sdpa --persistent_data_loader_workers ^
  --max_data_loader_n_workers 2 ^
  --seed 42 ^
  --gradient_checkpointing ^
  --mixed_precision bf16 ^
  --save_precision bf16 ^
  --network_module networks.lora_flux ^
  --network_dim 4 ^
  --optimizer_type adamw8bit ^
  --learning_rate 8e-4 ^
  --cache_text_encoder_outputs ^
  --cache_text_encoder_outputs_to_disk ^
  --fp8_base ^
  --highvram ^
  --max_train_epochs 6 ^
  --save_every_n_epochs 4 ^
  --dataset_config "C:\pinokio\api\fluxgym.git\outputs\sin-metalfont\dataset.toml" ^
  --output_dir "C:\pinokio\api\fluxgym.git\outputs\sin-metalfont" ^
  --output_name sin-metalfont ^
  --timestep_sampling shift ^
  --discrete_flow_shift 3.1582 ^
  --model_prediction_type raw ^
  --guidance_scale 1 ^
  --loss_type l2 ^
  --fp16_master_weights_and_gradients ^
  --full_fp16`

**Comment by joshgura** [comment]
> You can place your own models in the models folder, then edit the models.yaml and it will show up in the dropdown box.

tried editing the yaml and fluxgym didn't like it because it seems to only like models that have a huggingface repo, and in some cases even if there is a repo on huggingface, fluxgym might reject it. how do you skip the 'repo' and license and stuff and still have the model loadable?

**Comment by Summer-FF** [comment]
i wanna to know too

**Comment by thanatopsis70** [comment]
I'm using Pinokio for Fluxgym.
I commented out the "model.to(torch.float8_e4m3fn)" line in the flux_train_network.py file inside the sd-scripts folder.
I also commented out other stuff and added some notes just to remind me:

 ```
   if args.fp8_base:
            # check dtype of model
            if model.dtype == torch.float8_e4m3fnuz or model.dtype == torch.float8_e5m2 or model.dtype == torch.float8_e5m2fnuz:
                raise ValueError(f"Unsupported fp8 model dtype: {model.dtype}")
            elif model.dtype == torch.float8_e4m3fn:
                logger.info("Loaded fp8 FLUX model")
            else:
                logger.info("Skipping FP8 conversion. Keeping model in FP16.")
             #       "Cast FLUX model to fp8. This may take a while. You can reduce the time by using fp8 checkpoint."
             #       " / FLUXモデルをfp8に変換しています。これには時間がかかる場合があります。fp8チェックポイントを使用することで時間を短縮できます。"
             #   )
             #   model.to(torch.float8_e4m3fn)
```

I just re-trained a model on FP16 that I just did with the FP8 conversion. I haven't tested it enough to see if it is worth it, but it didn't take any longer to train the model. I have 24 GB of VRAM and 64GB RAM. The FP8 one took a few minutes less, but I also only output 1 sample pic every 4 epochs, and with the FP16 model I output 6 pics 4 every epochs. I don't recall the total steps and all, but they were the same, and both models finished in an hour.
